### ISSUE TITLE
feature request: attach an existing security group

### ISSUE THREAD

**Issue by okadakk** [aws/copilot-cli#1453]
Due to security requirements, we have a security group in place to limit the instances that can access the RDB.
I would like to be able to attach an existing security group.

Thank you!

**Comment by iamhopaul123** [comment]
Hello @okadakk, this is a very helpful feedback for us! Does it help if Copilot allows importing/configuring a security group on behalf of your service, so as to control the ingress/egress?

**Comment by efekarakus** [comment]
To add on to @iamhopaul123's comment, what would be an ideal UX for your usecase?

For example, if we add a `security_groups` field in the manifest would that help for adding additional security groups to your ECS service?
```yaml
cpu: 256
memory: 512
security_groups: # Additional security groups
  - sg-xxxxxxxx
```
Is specifying the the security group ID okay with you in the manifest file?

**Comment by okadakk** [comment]
Thanks @iamhopaul123 @efekarakus !

> Does it help if Copilot allows importing/configuring a security group on behalf of your service, so as to control the ingress/egress?

Yes! It helps me a lot.

> if we add a security_groups field in the manifest would that help for adding additional security groups to your ECS service?

Sounds good! It's OK specify in the manifest file!

**Comment by alex-all3dp** [comment]
We would also greatly benefit greatly from such a feature!

@efekarakus  Would it then also be possible to define custom security groups as an add-on and reference them in the manifest in order to being assigned to the load balancer?

**Comment by iamhopaul123** [comment]
Hello @alex-all3dp, why do you want to have security groups defined as addons (like what do you use them for)? Could you provide us more detailed use cases for that? Currently we already have a security group for the ecs service that allows traffic from load balancer and self.

**Comment by alex-all3dp** [comment]
@iamhopaul123 We would like to be able to modify the default security groups in order to disallow traffic from everywhere but rather restrict inbound traffic to a set of fixed IPs (Cloudflare in our case). Any chance that changing the default security groups would be possible in the near future?

Furthermore our setup would require us to also provide a certificate from ACM to the HTTPS listener on the load balancer.

**Comment by efekarakus** [comment]
Gotcha! That makes sense, for the first usecase we have this existing issue: https://github.com/aws/copilot-cli/issues/1234, which would be low effort for us to prioritize and deliver. Would that satisfy your requirement?

For the ACM cert and HTTPS listener, did you check out `copilot app init --domain` Copilot will do all that for you if you provide it with a domain name

**Comment by alex-all3dp** [comment]
@efekarakus Thank your for your feedback! 

Yes, I think the `allow-list` would satisfy our requirements for that use-case

Thanks for the hint, regarding the `--domain` flag, I'll give that a try!

**Comment by efekarakus** [comment]
Heya @alex-all3dp ! The allow-list feature was released in v0.6.0 https://github.com/aws/copilot-cli/releases/tag/v0.6.0 🚀 Hope this helps!

**Comment by alex-all3dp** [comment]
@efekarakus Thanks a lot! It's great how active you guys respond to feature requests by the community! :)

Btw. it seems that you forgot to mention it in the release notes and the official docs?

**Comment by efekarakus** [comment]
@alex-all3dp apologies, it's not super obvious in the docs: https://aws.github.io/copilot-cli/docs/manifest/lb-web-service/#http-allowed-source-ips

and in the release, it's the latest bullet point:
> * The `http` field for Load Balanced Web Services now supports `allowed_source_ips` to limit the IP addresses that can access your service. (#1643)

**Comment by alex-all3dp** [comment]
@efekarakus my bad, thx for the hint :)

**Comment by efekarakus** [comment]
Hi folks this is released in v1.3.0! https://github.com/aws/copilot-cli/releases/tag/v1.3.0

In your manifests you can do:
```yaml
name: 'very-fancy-service'
network:
  vpc:
    security_groups:
    - 'sg-1231414'
```

**Comment by mvn-bachhuynh-dn** [comment]
Hi @efekarakus ,
```
name: 'very-fancy-service'
network:
  vpc:
    security_groups:
    - 'sg-1231414'
 ```
This security group just apply for tasks not for ELB. How about I want to use existed security group for an ELB, which copilot created
